app/src/main/kotlin/at/bitfire/davdroid/push/UnifiedPushReceiver.kt

```diff
@@ -77,7 +84,20 @@ class UnifiedPushReceiver: MessagingReceiver() {
                 collectionRepository.getSyncableByTopic(topic)?.let { collection ->
                     serviceRepository.get(collection.serviceId)?.let { service ->
                         val account = accountRepository.fromName(service.accountName)
-                        syncWorkerManager.enqueueOneTimeAllAuthorities(account, fromPush = true)
+                        val syncDataType = if (collection.type == TYPE_ADDRESSBOOK) {
+                            SyncDataType.CONTACTS
+                        } else {
+                            SyncDataType.EVENTS
+                        }
+                        syncWorkerManager.enqueueOneTime(account, syncDataType, fromPush = true)
+
+                        // If the collection supports tasks, also schedule the tasks type
+                        if (collection.supportsVJOURNAL != false || collection.supportsVTODO != false) {
+                            // Make sure there's a tasks provider installed
+                            if (tasksAppManager.get().currentProvider() != null) {
+                                syncWorkerManager.enqueueOneTime(account, SyncDataType.TASKS, fromPush = true)
+                            }
+                        }
```

Comment:
I'd use a list with syncDataTypes to sync, and then handle

- type = address books
  - add `SyncDataType.CONTACTS`
- else (type = calendar):
  - add `SyncDataType.EVENTS`
  - check tasks app; if available, add `SyncDataType.TASKS`

And then enqueue a sync for all types of the list.

I think it's then easier to understand as when we mix address books and events and then handle tasks separately.